azkaban-db/src/main/java/azkaban/db/MySQLDataSource.java

```diff
@@ -43,12 +44,13 @@ public MySQLDataSource(final String host, final int port, final String dbName,
 
   /**
    * This method overrides {@link BasicDataSource#getConnection()}, in order to have retry logics.
+   * We don't make the call synchronized in order to guarantee normal cases performance.
    */
   @Override
-  public synchronized Connection getConnection() throws SQLException {
+  public Connection getConnection() throws SQLException {
 
     Connection connection = null;
-    int retryAttempt = 0;
+    int retryAttempt = 1;
     while (retryAttempt < AzDBUtil.MAX_DB_RETRY_COUNT) {
```

Comment:
retryAttempt <= AzDBUtil.MAX_DB_RETRY_COUNT

Reply:
Thanks for finding! I can not see much difference since we will never reach `MAX_DB_RETRY_COUNT`. IMO, it should not be a big deal